### PR TITLE
Added ProjectCodeTest that tests if all classes inside tests are internal and final or abstract

### PR DESCRIPTION
**Pull request by keradus**
I have treated [that comment](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1387#issuecomment-138674838) as a request!

ping @GrahamCampbell

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Nice. :)

File: Symfony/CS/Tests/ProjectCodeTest.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            }
+        );
+    }
+
+    private function registerAllProjectClasses()
+    {
+        $finder = Finder::create()
+            ->files()
+            ->in(__DIR__.'/..')
+            ->exclude(array(
```
Comment (SpacePossum) [inline Symfony/CS/Tests/ProjectCodeTest.php]: I would add `->name('*.php')` as well, so `readme`, `.php_cs`, etc. can still be added later on into the directories (when needed).

**Comment by SpacePossum** [comment]
Look good :)

File: Symfony/CS/Tests/ProjectCodeTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if (null === $classes && null === $projectClasses) {
+            $this->registerAllProjectClasses();
+            $projectClasses = $this->getClasses('Symfony\\CS\\', get_declared_classes());
+        }
+
+        if (null === $classes) {
+            $classes = $projectClasses;
+        }
+
+        $ns = $prefix;
```
Comment (stof) [inline Symfony/CS/Tests/ProjectCodeTest.php]: ``$ns`` is unused

**Comment by keradus** [comment]
Now test internal too ;)

**Pull request closed without merging**